CE-47 Metrics missing from documentation (#1604)
hashicorp/web-unified-docs: content/consul/v1.22.x/content/docs/monitor/telemetry/agent.mdx

|[Prometheus](https://vector.dev/docs/reference/configuration/sinks/prometheus_exporter/#flush_period_secs)| 60s|
|[statsd](https://github.com/statsd/statsd/blob/master/docs/metric_types.md#timing)|10s|

The telemetry information can be used for debugging or otherwise getting a better view of what Consul is doing. 

To retrieve telemetry data, send a signal to the Consul process: on Unix, this is `USR1` while on Windows it is `BREAK`. 

> Once Consul receives the signal, it will dump the current telemetry information to the agent's `stderr`.

Reviewer: ```suggestion
To retrieve telemetry data, send a signal to the Consul process: 

- On Unix, send `USR1` 
- On Windows, send`BREAK`. 

When Consul receives the signal, it dumps the current telemetry information to the agent's `stderr`. The output resembles the following example:
```

[G] 'consul.autopilot.failure_tolerance': 0.000
[G] 'consul.state.config_entries.dc1.mesh': 0.000
[G] 'consul.runtime.num_goroutines': 335.000
[G] 'consul.raft.commitNumLogs': 1.000
[G] 'consul.state.connect_instances.dc1.terminating-gateway': 0.000
[G] 'consul.state.connect_instances.dc1.connect-proxy': 4.000
[G] 'consul.state.kv_entries.dc1': 0.000
[G] 'consul.state.config_entries.dc1.service-router': 0.000
[G] 'consul.state.config_entries.dc1.service-resolver': 0.000
[G] 'consul.state.config_entries.dc1.service-splitter': 0.000
[G] 'consul.runtime.free_count': 1809554.000
[G] 'consul.runtime.heap_objects': 150825.000
[G] 'consul.raft.leader.dispatchNumLogs': 1.000
[G] 'consul.server.isLeader': 1.000
[G] 'consul.state.services.dc1': 10.000
[G] 'consul.state.connect_instances.dc1.api-gateway': 1.000
[G] 'consul.state.config_entries.dc1.service-intentions': 4.000
[G] 'consul.state.config_entries.dc1.terminating-gateway': 0.000
[G] 'consul.runtime.sys_bytes': 67721480.000
## ...
```

</CodeBlockConfig>

The information can also be retrieved using with the [metrics endpoint](/consul/api-docs/agent#view-metrics) in JSON format or [Prometheus](https://prometheus.io/) format.

> By default, all metric names of gauge type are prefixed with the hostname of the Consul agent, e.g., `consul.hostname.server.isLeader`. 

Reviewer: ```suggestion
By default, all metric names of gauge type use the hostname of the Consul agent as a prefix. For example,  `consul.<hostname>.server.isLeader`. 
```

To disable prefixing the hostname, set `telemetry.disable_hostname=true` in the [agent configuration](/consul/docs/reference/agent/configuration-file/telemetry).

Additionally, if the [`telemetry` configuration options](/consul/docs/reference/agent/configuration-file/telemetry)
are provided, the telemetry information will be streamed to a [statsite](http://github.com/armon/statsite) or [statsd](http://github.com/etsy/statsd) server where it can be aggregated and flushed to Graphite or any other metrics store.

A [Grafana dashboard](https://grafana.com/grafana/dashboards/13396), maintained by the Consul team, is also available. The dashboard helps display these metrics for easy visualization and can be used as a starting point to create your own dashboards.

> For hands-on examples review:

Reviewer: ```suggestion
The following documentation and tutorials provide additional examples of the telemetry configuration process:
```

- the [Monitor Consul datacenter health with Telegraf](/consul/docs/monitor/telemetry/telegraf) documentation
- the [Monitor Consul server health and performance with metrics and logs](/consul/tutorials/observe-your-network/server-metrics-and-logs) tutorial
- the [Observe Consul service mesh traffic](/consul/tutorials/get-started-vms/virtual-machine-gs-monitoring) VMs tutorial
- the [Observe Consul service mesh traffic](/consul/tutorials/get-started-kubernetes/kubernetes-gs-observability) Kubernetes tutorial

## Key Metrics

> These are some metrics emitted that can help you understand the health of your cluster at a glance. 

Reviewer: ```suggestion
These are some of the most important metrics that can help you understand the health of your cluster at a glance. 
```

### License Expiration <EnterpriseAlert inline />

| Metric Name                       | Description                                                      | Unit  | Type  |
| :-------------------------------- | :--------------------------------------------------------------- | :---- | :---- |
| `consul.system.licenseExpiration` | Number of hours until the Consul Enterprise license will expire. | hours | gauge |

**Why they're important:** This measurement indicates how many hours are left before the Consul Enterprise license expires. When the license expires some Consul Enterprise features will cease to work. An example of this is that after expiration, it is no longer possible to create or modify resources in non-default namespaces or to manage namespace definitions themselves even though reads of namespaced resources will still work.

**What to look for:** This metric should be monitored to ensure that the license doesn't expire to prevent degradation of functionality.

### WAL logstore performance

The WAL LogStore replaces the BoltDB backend used in previous versions of Consul. 

> Refer to [Experimental WAL LogStore backend](/consul/docs/deploy/server/wal/) for more information.

Reviewer: ```suggestion
The WAL LogStore replaced BoltDB as the backend used in previous versions of Consul. 
Refer to [Persistent data backend architecture](/consul/docs/architecture/backend) for more information.
```

| Metric Name                                | Description                                                                     | Unit  | Type  |
| :----------------------------------------- | :------------------------------------------------------------------------------ | :---- | :---- |

> | `consul.raft.wal.head_truncations`         | Counts how many log entries have been truncated from the head - i.e. the oldest entries. By graphing the rate of change over time you can see individual truncate calls as spikes                                               | log entries        | counter |

Reviewer: ```suggestion
| `consul.raft.wal.head_truncations`         | Counts how many log entries have been truncated from the head, which represents the oldest entries. By graphing the rate of change over time you can observe individual truncate calls as spikes                                               | log entries        | counter |
```
